src/app/components/BlockStats/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { FC, ReactNode } from 'react'
+import Box from '@mui/material/Box'
+import Tooltip from '@mui/material/Tooltip'
+import { styled } from '@mui/material/styles'
+import { COLORS } from '../../../styles/theme/colors'
+
+const StyledSquare = styled(Box, {
+  shouldForwardProp: prop => prop !== 'success',
+})(({ success }: { success?: boolean }) => {
+  return {
+    display: 'flex',
+    width: '24px',
+    height: '24px',
+    borderRadius: '3px',
+    backgroundColor: success ? COLORS.eucalyptus : COLORS.errorIndicatorBackground,
+  }
+})
+
+type BlockStatsProps<T> = {
+  data: T[]
+  dataKey: Extract<keyof T, string>
+  legendLabels?: {
+    success: string
+    fail: string
+  }
+  tooltipFormatter?: (data: string) => ReactNode
+}
+
+export const BlockStats: FC<BlockStatsProps<any>> = <T extends { [key: string]: any }>({
```

Comment:
```suggestion
export const BlockStats = <T extends { [key: string]: any }>({
```

`FC<BlockStatsProps<any>>` just removes typesafety on dataKey